remote-query/remote-query-server/src/main/java/org/infinispan/query/remote/impl/LazySearchMapping.java

```diff
@@ -126,6 +129,18 @@ public void reload() {
       }
    }
 
+   @Override
+   public void restart() {
+      long stamp = stampedLock.writeLock();
```

Comment:
Locking is a good idea IMO, but I don't think the query interceptor holds the lock for the whole duration of indexing? I.e. it seems to me that this sequence of events would fail:

* `QueryInterceptor` gets the mapping (successfully gets a read lock, then releases it)
* `restart` is called (successfully gets a write lock). Hibernate Search performs the restart, at least to the point where it stops the older mapping/integration.
* `QueryInterceptor` performs indexing => Failure, it's using the old mapping, which has been stopped.

Reply:
Yes the lock here is just to avoid concurrent restart / reload.
There is no lock at the moment on the query interceptor, so if the restart is progress a log message warning the user to not write on caches during the restart.